README-LINUX.md

```diff
@@ -0,0 +1,148 @@
+```
+
+If installed as a system package, the files will be found in the following
+locations:
+
+```
+$PREFIX/share/applications/com.mitchellh.ghostty.desktop
+$PREFIX/share/dbus-1/services/com.mitchellh.ghostty.service
+$PREFiX/lib/systemd/user/com.mitchellh.ghostty.service
+```
+
+Ghostty takes advantage of D-Bus activation and `systemd` user services to provide
+the best experience possible.
+
+The application launcher, instead of executing Ghostty directly, will send a
+signal to the D-Bus session broker asking it to open a Ghostty window. The
+D-Bus session broker will check D-Bus to see if Ghostty is already running. If
+Ghostty is already running, the broker will send a D-Bus signal to Ghostty to
+create a new window.
+
+```mermaid
+sequenceDiagram
+  actor Launcher
+  participant D-Bus
+  participant Ghostty
+  Launcher->>D-Bus: Start a Ghostty window
+  D-Bus->>Ghostty: Create a new window
+  Ghostty->>D-Bus: OK
+```
+
+If Ghostty is not running, the D-Bus session broker will ask `systemd` to start
+Ghostty in a user service, and then will send Ghostty a D-Bus signal to create a
+new window.
+
+```mermaid
+sequenceDiagram
+  actor Launcher
+  participant D-Bus
+  participant systemd
+  Launcher->>D-Bus: Start a Ghostty window
+  D-Bus->>systemd: Start Ghostty service
+  create participant Ghostty
+  systemd->>Ghostty: Start Ghostty service
+  Ghostty->>systemd: Started
+  Ghostty->>D-Bus: I'm here!
+  D-Bus->>Ghostty: Create a new window
+  Ghostty->>D-Bus: OK
+```
+
+This ensures that any startup delay caused by GTK (or any other initialization) is
+only incurred once per login.
+
+The Ghostty `systemd` user service can be configured to start up as soon as you
+log in by running this command:
+
+```sh
+systemctl enable --user com.mitchell.ghostty.service
+```
+
+By doing so any startup delay will be "hidden" by the time that your system
+needs to finish setting up everything after logging in.
+
+## Stopping Ghostty
+
+The Ghostty `systemd` user service can be stopped by running:
+
+```sh
+systemctl stop --user com.mitchellh.ghostty.service
+```
+
+> [!WARNING]
+>
+> There is a chance of data loss if the above command is run while there are any
+> open Ghostty windows with programs running in them. Any programs running in
+> Ghostty windows will be killed without asking for any confirmation.
+
+## Logs
+
+Any logs emitted by the Ghostty `systemd` user service can be seen by running
+this command:
+
+```sh
+journalctl -a -f --user -u com.mitchellh.ghostty.service
+```
+
+## Creating New Windows
+
+With D-Bus activation and `systemd` user services in use, simply launching
+`ghostty` from the CLI (or through a global keybind) will not create a new
+Ghostty terminal window. The following command can be used to send a D-Bus
+signal to Ghostty to create a new window:
+
+```sh
+gdbus call --session --dest com.mitchellh.ghostty --object-path /com/mitchellh/ghostty --method org.gtk.Actions.Activate new-window [] []
+```
```

Comment:
This seems like something that the Ghostty CLI could handle. Is that a wrong assumption?

Reply:
How did you install? The service files need to be in the right locations in `~/.local` for this to work.

Reply:
> This seems like something that the Ghostty CLI could handle. Is that a wrong assumption?

It wouldn't be hard to add a `+gtk-new-window` CLI action that sends the D-Bus signal, but that might be too GTK specific for a CLI action.  Doing something deeper would require rethinking the GTK app init code and would take some deeper thinking.

Reply:
> How did you install? The service files need to be in the right locations in `~/.local` for this to work

That's what I did.

Reply:
Hmm, I'd check the files to see if they are correct. Also, have you logged in/out recently, or run `systemctl daemon-reload --user`? Maybe `systemd` hasn't picked up the new files.